ui/components-react/src/test/propertygrid/component/internal/FlatPropertyRenderer.test.tsx

```diff
@@ -168,7 +168,107 @@ describe("FlatPropertyRenderer", () => {
     );
   });
 
-  it("renders array as a non primitive value", () => {
+  it("renders array as a non primitive value when renderer assigned but not registered", () => {
+    propertyRecord = TestUtils.createArrayProperty("StringArray", [
+      TestUtils.createPrimitiveStringProperty("Label", "Model"),
+    ]);
+    propertyRecord.property.renderer = { name: "nonRegisteredRenderer" };
+
+    render(
+      <FlatPropertyRenderer
+        orientation={Orientation.Horizontal}
+        propertyRecord={propertyRecord}
+        isExpanded={false}
+        onExpansionToggled={() => {}}
+      />
+    );
+
+    expect(screen.getByTitle("StringArray (1)")).satisfy(
+      selectorMatches(
+        [
+          ".components-nonprimitive-property-label-renderer",
+          ".components-property-label-renderer",
+        ].join(" ")
+      )
+    );
+  });
+
+  it("renders array as a primitive value when a custom property renderer is registered", () => {
```

Comment:
I guess it would make more sense to check that custom renderer is used instead of checking for `.components-primitive-property-label-renderer`. Same for struct properties.

Reply:
I think same changes should be applied when testing custom rendering based on `typename`.

Reply:
Not sure how I missed them, fixed.